File: implementations/java-springboot/harness/test/testdata/utc-timestamp-source/good/account/infrastructure/persistence/AccountJpaEntity.java
package com.example.accountservice.account.infrastructure.persistence;

import com.example.accountservice.common.UtcClock;
import java.time.LocalDateTime;

public class AccountJpaEntity {
    private LocalDateTime touchedAt = UtcClock.now();

    LocalDateTime getTouchedAt() {
        return touchedAt;
    }
}

File: implementations/kotlin-springboot/harness/test/testdata/utc-timestamp-source/good/src/main/kotlin/account/infrastructure/persistence/AccountJpaEntity.kt
package com.example.accountservice.account.infrastructure.persistence

import java.time.LocalDateTime
import java.time.ZoneOffset

// Converts inline instead of going through a shared helper — still a UTC reading, so the rule
// accepts it (an external project need not have adopted a common clock package).
class AccountJpaEntity(
    var accountId: String = "",
    var createdAt: LocalDateTime = LocalDateTime.now(ZoneOffset.UTC),
)
